lib/streamlit/elements/deck_gl_json_chart.py

```diff
@@ -100,9 +101,15 @@ def dg(self) -> "streamlit.delta_generator.DeltaGenerator":
 
 # Map used when no data is passed.
 EMPTY_MAP: Dict[str, Any] = {
-    "initialViewState": {"latitude": 0, "longitude": 0, "pitch": 0, "zoom": 1}
+    "initialViewState": {"latitude": 0, "longitude": 0, "pitch": 0, "zoom": 1},
 }
 
+# Override the default value of map_style from 'dark' to None
+defaults = list(pdk.Deck.__init__.__defaults__)
+if defaults[2] == "dark":
+    defaults[2] = None
+    pdk.Deck.__init__.__defaults__ = tuple(defaults)
+
```

Comment:
@LukasMasuch Following from my previous comment of distinguishing between when users set the "dark" style vs pydeck setting "dark" by default: as there was no signal to tell us who had set the style, I override the default to be `map_style=None`. On the frontend, it translates to `mapStyle: undefined`.  

The `mapStyle: undefined` is our signal on the frontend that the user has not set `map_style` on the backend, which we then overwrite based on the Streamlit theme. If the user sets `map_style`, we respect it and don't overwrite the value on the frontend.

Is there a more elegant approach to tell if `map_style` was set by the user or used the default value?

Reply:
that is a clever solution 😄 However, I'm not sure if we really should overwrite API internals for this case... doing this kind of monkey patching should be a last resort. I'm leaning towards just setting it to `None` in the [marshall function](https://github.com/streamlit/streamlit/blob/defcf903684723b95a1f1c112fa628e156d786fb/lib/streamlit/elements/deck_gl_json_chart.py#L119) whenever `dark` is used. This will slightly break the usage of `dark` as explicit map style, however all the other map styles should be functional. Even dark theme via the full mapbox style URL could be used in this case, or?

Reply:
Or is the map style already transformed to the full mapbox URL at that point in the marshall function?

Reply:
>Or is the map style already transformed to the full mapbox URL at that point in the marshall function?

Correct. By the time the `pydeck_obj` is passed to the marshall function, pydeck has already transformed the default `map_style="dark"` into the full mapbox URI:

```
https://basemaps.cartocdn.com/gl/dark-matter-gl-style/style.json
```

Reply:
> However, I'm not sure if we really should overwrite API internals for this case... doing this kind of monkey patching should be a last resort.

Agreed! I too am uncomfortable with this monkey patching. Unsure how to proceed 🤔

Reply:
@snehankekre What happens if you set it manually to something like: `mapbox://styles/mapbox/dark-v9`? Is it also transformed into a full URL?

Reply:
In the marshall function or when setting `map_style` in `pdk.Deck()`?

Reply:
> By the time the pydeck_obj is passed to the marshall function
So it's already transformed in the `pydeck_obj` and not by the `to_json()` call?

Reply:
> In the marshall function or when setting map_style in pdk.Deck()?
probably both. In the marshall function we are just getting the pydeck object in.. from the `st.pydeck` command.. so that's all we can work with

Reply:
Yeah it's set in `pdk.Deck` [here](https://github.com/visgl/deck.gl/blob/master/bindings/pydeck/pydeck/bindings/deck.py#L100), meaning it's already transformed in the `pydeck_obj` and not by the `to_json()` call.

When I `print(pydeck_obj)` I see `"mapStyle": "https://basemaps.cartocdn.com/gl/dark-matter-gl-style/style.json"`

Reply:
> What happens if you set it manually to something like: mapbox://styles/mapbox/dark-v9

When set in `pdk.Deck()`, the console says `mapStyle: mapbox://styles/mapbox/dark-v9`. It remains unchanged.

Reply:
ok, I think I have an idea of how we approach this. We have [a project](https://www.notion.so/streamlit/Beautiful-default-styles-for-3rd-party-charting-libraries-a7d6bf5f45a0485bb178b64643b85f87) to add a `theme` parameter to our charting commands that activates the streamlit theme as default, but also allows you to deactivate it. `st.pydeck` might be a good candidate to add this variable as well, and if streamlit theme is selected we set the `mapstyle` to `None`. For now I would push it without setting something to `None`, so the way to use it in `st.pydeck` would be only be setting it explicitly to `None`.

Reply:
Whereas `theme=streamlit` would mean that the styling/theming integrates well with Streamlit.

Reply:
Ok, IIUC I had to just undo the monkey patching for now? If so, fixed on b4a4b7a

Reply:
yep, I think so. If tests run through we can merge it :) Maybe also add a short text to the [project](https://www.notion.so/streamlit/Beautiful-default-styles-for-3rd-party-charting-libraries-a7d6bf5f45a0485bb178b64643b85f87) about `st.pydeck` and why it might benefit from the `theme` parameter as well (so we keep track of this)